snapshot/read_write_hdf.py

```diff
@@ -0,0 +1,130 @@
+                        )
+                    # If dataset exists resize and add current data
+                    else:
+                        main_file[key].resize(main_file[key].shape[0] + 1, axis=0)
+                        main_file[key][-1] = current_data[i]
+            parameter_file.close()
+        main_file.close()
+
+    def write_sim_output_to_hdf(
+        self, file_path: str, macro_data: dict, micro_data: dict
+    ) -> None:
+        """
+        Write the output of a micro simulation to a hdf5 file.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to file in which the data should be written.
+        macro_data : dict
+            Dict of macro simulation input.
+        micro_data : dict
+            Dict of micro simulation output.
+        """
+        parameter_file = h5py.File(file_path, "a")
+        input_data = [macro_data, micro_data]
+        # Iterate over macro and micro data sets
+        for i in range(len(input_data)):
+            # Iterate over all keys in the current data set
+            for key in input_data[i].keys():
+                current_data = np.asarray(input_data[i][key])
+                # If dataset corresponding to dictionary key does not exist create and add current data
+                if key not in parameter_file.keys():
+                    parameter_file.create_dataset(
+                        key,
+                        data=current_data,
+                        shape=(1, *current_data.shape),
+                        maxshape=(None, *current_data.shape),
+                    )
+                    parameter_file[key]
+                # If dataset exists resize and add current data
+                else:
+                    parameter_file[key].resize(parameter_file[key].shape[0] + 1, axis=0)
+                    parameter_file[key][-1] = current_data
+        parameter_file.close()
+
+    def read_hdf_to_dict(self, file_path: str, data_names: dict) -> list:
+        """
+        Read data from a hdf5 file and return it as a list of dictionaries.
+
+        Parameters
+        ----------
+        file_path : str
+            Path of file to read data from.
+        data_names : dict
+            Names of parameters to read from the file.
+
+        Returns
+        -------
+        list
+            List of dicts where the keys are the names of the parameters and the values the corresponding data.
+        """
+
+        parameter_file = h5py.File(file_path, "r")
+        parameter_data = dict()
+        output = []
+        # Read data by iterating over the relevant datasets
+        for key in data_names.keys():
+            parameter_data[key] = np.asarray(parameter_file[key][:])
+            my_key = key
+        # 3. iterate over len of data. In each iteration write data from all macro data sets (using the read_data_names as keys) to a dictionary with the key being the name of the data set
```

Comment:
Review comments.

Reply:
Any input required here? Shortening the comment would be good.